CONTRIBUTING.md

```diff
@@ -55,9 +55,16 @@ conda install pytorch -c pytorch-nightly
 pip install cmake ninja
 ```
 
+### Install other requirements
+
+```bash
+pip install -r requirements.txt
+```
+
 ```bash
 git clone https://github.com/pytorch/audio.git
 cd audio
+git submodule sync
```

Comment:
I do not think this is necessary.
Our CI just works without it https://github.com/pytorch/audio/blob/7deea25948383fdaad63893b7f5676e06fd8a5ad/.circleci/unittest/linux/scripts/install.sh#L44

Reply:
Both @yangarbiter and I just ran into compilation errors with kaldi despite running submodule update on fresh clone. @yangarbiter noticed we simply needed to sync. Since we are reminding users to both clone and update submodule, it seems like we might as well remind them to sync. @mthrok, have you ran seen [this error](https://github.com/pytorch/audio/files/6677554/buffer.txt)?

Reply:
`pip` cannot be used for source build. Please follow the documentation and use `python setup.py develop`.

Reply:
Note: the proper way to change the build process so that one can perform `pip install` is to perform submodule update in somewhere `setup.py`. That way, `pip install git+https://github.com/pytorch/audio` will work. 
ref: https://github.com/pytorch/text/pull/1154

Reply:
I thought `git submodule update` and `git submodule sync` are doing slightly different things ([(link)](https://stackoverflow.com/questions/45678862/git-submodule-update-vs-git-submodule-sync/45679261) ). It appears that if you just pull from the current master then it should be enough to do only `git submodule update`. But if you `git submodule update` a while ago and the url for the submodule have changed, then you'll need to do `git submodule sync` to update the url.

Reply:
Yeah, `sync` is necessary when there is a change on submodule. But here, the context is for first time, as the explanation starts from `git clone`.

How about adding some comments when you need it?

```bash
git submodule sync  # Only when submodule URLs are changed upstream.
```

Reply:
> `pip` cannot be used for source build. Please follow the documentation and use `python setup.py develop`.

The issue occurs with `python setup.py develop` too, see [buffer.txt](https://github.com/pytorch/audio/files/6678092/buffer.txt).

> How about adding some comments when you need it?
> 
> ```shell
> git submodule sync  # Only when submodule URLs are changed upstream.
> ```

SGTM

Reply:
I think this is due to something other than submodule. Looking at the log Kaldi is pulled correctly. The submodule has not been updated.

Reply:
We don't have concrete evidence showing `git submodule sync` fixes this bug. I'll remove this line from README.